<?xml version="1.0"?>

<!--

This launch file starts a node from the Youbot-RVO package.
This initialises a real experiment, controlling multiple Youbots with hrvo/inference.

If you receive an error message that contains "Excecute as root"
then go to the bin folder and execute: sudo setcap cap_net_raw+ep
or enable the USE_SETCAP flag in the cmake file and recompile again.
-->

<launch>
        <!-- Start the experiment node -->
        <node name="youbot_experiment" pkg="hrvo" type="Experiment" output="screen" clear_params="true">
                <!-- Robot Setup -->
                <rosparam file="$(find hrvo)/cfg/robot.yaml" command="load" />

                <!-- Experiment Setup -->
                <rosparam file="$(find hrvo)/cfg/experiment.yaml" command="load" />

                <!-- Logger Parameters -->
                <rosparam file="$(find hrvo)/cfg/logger.yaml" command="load" />

                <!-- Console Display -->
                <rosparam file="$(find hrvo)/cfg/console.yaml" command="load" />

                <!-- Constants / DO NOT CHANGE! -->
                <rosparam file="$(find hrvo)/cfg/constants.yaml" command="load" />

                <!-- Model Parameters -->
                <rosparam file="$(find hrvo)/cfg/model.yaml" command="load" />

                <!-- Workspace limits -->
                <rosparam file="$(find hrvo)/cfg/workspace.yaml" command="load" />

                <!-- Simulation Parameters -->
                <rosparam file="$(find hrvo)/cfg/simulation.yaml" command="load" />

                <!-- Remapping -->
                <remap from="prime/cmd_vel" to="base_controller/command"/>
                <remap from="prime/odom" to="base_odometry/odom" />
                <remap from="prime/amcl_pose" to="amcl_pose" />

                <!-- Temporary settings -->
                <!-- <remap from="youbot_2/odom" to="odom" /> -->
                <!-- <remap from="youbot_2/cmd_vel" to="move_base/cmd_vel" /> -->
                <!-- <remap from="youbot_2/cmd_vel" to="cmd_vel" /> -->
                <!-- <remap from="youbot_2/amcl_pose" to="amcl_pose" /> -->
        </node>

</launch>
